### PR TITLE
feat: add an auth endpoint to update the user's password

### PR DESCRIPTION
**Pull request by tiescl**

### CODE REVIEW

File: src/modules/auth/auth.service.ts [hunk 1]
```diff
@@ -61,6 +63,29 @@ export class AuthService {
...
+    const isPasswordValid = await Hasher.verifyHash(
+      user.passwordHash,
+      dto.oldPassword,
+    );
+    if (!isPasswordValid) {
+      throw new BadRequestException('Invalid email or password');
+    }
+
+    const hashedPassword = await Hasher.hashValue(dto.newPassword);
+
```
Comment (farrooh) [inline src/modules/auth/auth.service.ts:76]: (Optional) You can also validate that the new password is not the same as the old password. Or maybe it's done in frontend.

**Review by farrooh — CHANGES_REQUESTED** [review]

**Review by purple-mountain — APPROVED** [review]
you can also add swagger docs for your new route. you don't have to write unit tests, I have already written them for auth module in my branch.

File: src/modules/auth/auth.controller.ts [hunk 2]
```diff
@@ -70,6 +72,31 @@ export class AuthController {
     res.clearCookie('refreshToken');
   }
 
+  @Post('update-password')
+  @UseGuards(AuthTokenGuard)
+  async updatePassword(
+    @GetUser() user: User,
+    @Body() registerDto: UpdatePasswordDto,
```
[the remark this answers is not in the document]
Comment (tiescl) [reply src/modules/auth/auth.controller.ts:79]: fixed

File: src/modules/auth/auth.controller.ts [hunk 3]
```diff
@@ -70,6 +72,31 @@ export class AuthController {
...
+        this.configService.get<string>('AUTH_ACCESS_TOKEN_EXPIRES_IN'),
+      ),
+    });
+    res.cookie('refreshToken', tokens.refreshToken, {
+      httpOnly: true,
+      secure: isProd(this.configService.get('NODE_ENV')),
+      maxAge: ms(
+        this.configService.get<string>('AUTH_REFRESH_TOKEN_EXPIRES_IN'),
+      ),
+    });
```
[the remark this answers is not in the document]
Comment (tiescl) [reply src/modules/auth/auth.controller.ts:95]: fixed

File: src/modules/auth/auth.service.ts [hunk 4]
```diff
@@ -61,6 +63,29 @@ export class AuthService {
     return { accessToken, refreshToken };
   }
 
+  async updatePassword(dto: UpdatePasswordDto, user: User) {
+    const isPasswordValid = await Hasher.verifyHash(
+      user.passwordHash,
+      dto.oldPassword,
+    );
+    if (!isPasswordValid) {
+      throw new BadRequestException('Invalid email or password');
```
[the remark this answers is not in the document]
Comment (tiescl) [reply src/modules/auth/auth.service.ts]: fixed

File: src/modules/auth/auth.service.ts (hunk 1 above)
Comment (tiescl) [reply src/modules/auth/auth.service.ts:76]: handled in frontend

**Comment by purple-mountain** [comment]
you should also do the same for Oauth with cookies util. I think it is almost identical

**Review by farrooh — APPROVED** [review]

**Pull request merged**